frontend/instance/frontend/init-instance.sh

```diff
@@ -10,6 +10,19 @@ aws --region $region s3 cp --recursive s3://membership-dist/${stack}/${stage}/fr
 mkdir /etc/gu
 aws --region $region s3 cp s3://membership-private/${stage}/membership.private.conf /etc/gu
 
+# wait for dpkg lock to become free
+lsof /var/lib/dpkg/lock
+rc=$?
+echo "Return code is $rc"
+until [ "$rc" -eq "1" ]; do
+  echo "Waiting for dpkg lock to become free"
+  sleep 1s;
+  lsof /var/lib/dpkg/lock
+  rc=$?
+  echo "Return code is $rc"
+done
+echo "dpkg lock is free, ready to install application"
+
 # install and start play app and create frontend user
 dpkg -i /dist/frontend_1.0-SNAPSHOT_all.deb
```

Comment:
agreed thanks for the comments, we checked the lock status for the first 30 seconds from when the userdata started, second 0 was held by one process, second 1-9ish was held by the unattended-update process, and the remaining time it was free all the time.  Another approach would have been to keep retrying dpkg inside the loop, instead of lsof, but it seemed neater to just try once.

aptdcon is an interesting one, although in our case we have to wait for the package to install before we can do some chown with the user that's from the package, so we'd have to block until it was installed.  But that's not to say any of the stuff we're doing is best practice!